templates/assets/js/assesment_v2.js

```diff
@@ -0,0 +1,124 @@
+const quizContainer = document.getElementById("quiz");
+const resultsContainer = document.getElementById("results");
+const submitButton = document.getElementById("submit");
+
+const easyQuestions = [];
+const mediumQuestions = [];
+const hardQuestions = [];
+const difficulty_dict = {
+  easy: easyQuestions,
+  medium: mediumQuestions,
+  hard: hardQuestions,
+  all: myQuestions,
+};
+let difficulty = "all";
+
+const populate_questions = () => {
+  let num = 0;
+  myQuestions.forEach((currentQuestion) => {
```

Comment:
Hard-coding the difficulty level names doesn't look good. Instead try to use either an enum ([enumify](https://github.com/rauschma/enumify)) or a simple array and loop through it so the code repetition can be taken care of. The different arrays then can be all moved into a map keyed by the name of the difficulty level.
You can use [lodash Groupby function](https://lodash.com/docs#groupBy) for the same effect.